src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -27,6 +28,8 @@ internal class ProjectHotReloadSessionManager : OnceInitializedOnceDisposedAsync
         private HotReloadState? _pendingSessionState = null;
         private int _nextUniqueId = 1;
 
+        public bool HasActiveHotReloadSessions => _activeSessions.Count != 0;
```

Comment:
>Count [](http://example.com/codeflow?start=66&length=5)

Since this isn't a concurrent dictionary, count isn't thread-safe. Does this matter in this case? Is this collection modified from multiple background tasks simultaneously?

Reply:
Looking below, you managed the modification with the _semaphore semaphore. Do you care that there is potentially races on this count?

Reply:
I looked at the source for Dictionary.Count and in the way I am using it I think it is fine since I am just checking against 0.  This dictionary is modified when a hot reload session starts and ends.  I could wait on the lock to ensure it is not modified, but I am not sure it matters in this scenario since updates to the dictionary should be relatively infrequent.

Reply:
Sounds good. Thanks for checking.